### ISSUE TITLE
Using ! or $spoolfile in mailboxes causes an empty folder view

### ISSUE THREAD

**Issue by thomergil** [neomutt/neomutt#1929]
## Expected Behaviour

Start neomutt, load an IMAP inbox.

## Actual Behaviour

Neomutt starts, but rather than connecting once, it seems (based on status messages) to connect to the IMAP server twice in a row resulting in an empty folder view.

## Steps to Reproduce

Configure an IMAPS account. (In case it's relevant: running dovecot on the server side.) I am unable to reproduce this problem with similar settings for a Gmail account.

Local neomutt configuration:

```
set folder="imaps://server.hostname/"
set spoolfile="imaps://server.hostname/"
mailboxes $spoolfile other1 other2
```

The problem can be "fixed" either by setting `folder` to empty, or by removing the `mailboxes` setting.

## How often does this happen?

- Always

## When did it start to happen?

- After upgrading to 20191102 from 20180716.

## NeoMutt Version

```
NeoMutt 20191102
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 18.7.0 (x86_64)
ncurses: ncurses 5.7.20081102 (compiled with 5.7.20081102)
libiconv: 1.11
libidn: 1.35 (compiled with 1.35)
GPGme: 1.13.1
hcache backends: tokyocabinet, lmdb

Compiler:
Apple clang version 11.0.0 (clang-1100.0.33.8)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

Configure options: --prefix=/usr/local/Cellar/neomutt/20191102 --enable-gpgme --with-gpgme=/usr/local/opt/gpgme --gss --lmdb --notmuch --sasl --tokyocabinet --with-ssl=/usr/local/opt/openssl@1.1 --with-ui=ncurses

Compilation CFLAGS: -g -O2 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/local/opt/gpgme/include -I/usr/local/opt/libassuan/include -I/usr/local/opt/libgpg-error/include -I/usr/local/Cellar/neomutt/20191102/include -DNCURSES_WIDECHAR -I/usr/local/opt/openssl@1.1/include

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn -inotify 
  -locales_hack -lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/local/Cellar/neomutt/20191102/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/usr/local/Cellar/neomutt/20191102/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>

```

**Comment by flatcap** [comment]
Sorry about that, @thomergil.
I committed a change yesterday, 6a0e1814a, that should fix this problem.
(it's likely there will be another release at the end of the week).

**Comment by thomergil** [comment]
@flatcap . The version I have this problem with includes this commit. The problem is still there.

**Comment by thomergil** [comment]
@flatcap, I also experience this problem with a freshly compiled version that includes this patch. Exactly the same behavior.

**Comment by flatcap** [comment]
Sorry, I can't repeat this.

Config:
```
set folder = "imap://apple@flatcap.org/"
set spoolfile = +inbox

mailboxes $spoolfile
mailboxes +abbot
mailboxes +achebe
mailboxes +awlinson
```

I see the four mailboxes in the sidebar.
When I change to the browser <kbd>c</kbd><kbd>?</kbd>  I see the **Mailboxes** view
```
1 +inbox
2 +abbot
3 +achebe
4 +awlinson
```

When I hit <kbd>\<Tab\></kbd>  I see the **Directory** view (listing all the folders)
```
1 IMAP INBOX
2 IMAP abbot
3 IMAP achebe
4 IMAP addams
5 IMAP alighieri
6 IMAP appleton
7 IMAP arnason
8 IMAP arundel
9 IMAP awlinson
```

**Comment by thomergil** [comment]
I figured out the problem. This is the same problem as [https://github.com/neomutt/neomutt/issues/1887](https://github.com/neomutt/neomutt/issues/1887). If `mailboxes` contains a non-existent mailbox, things break. Obviously commenting out `mailboxes` would fix that from happening. What I don't understand--but maybe it doesn't matter--is why commenting out `set folder` fixes it. Either way, thank you.

**Comment by flatcap** [comment]
Ah, OK.
There's a workaround, described here: https://github.com/neomutt/neomutt/issues/1901#issuecomment-547676531

**Comment by thomergil** [comment]
@flatcap , I don't actually think that workaround works for the problem I described: in my case the index view shows an empty inbox and I cannot switch between folders. It's a critical error, not just an inconvenience. I think `mailboxes` should gently tolerate non-existing folders.

**Comment by flatcap** [comment]
>  I think `mailboxes` should gently tolerate non-existing folders

Hmm... I'm still in the middle of some large changes to how the Mailboxes work.

Mutt maintains a large list of Mailboxes with no understanding of _what_ they are or _who_ owns them.

NeoMutt now groups the Mailboxes into Accounts.  These changes have meant I've had to be stricter when parsing `mailboxes`.  I _could_ restore the old behaviour, but I'd rather not. (but given the trouble I've caused a lot of users, I may have to).

In the near future, you will be able to **name** Accounts and the sidebar will understand how to group Accounts/Mailboxes.  This will allow the sidebar to make intelligent decisions on how to display things.

Soon...
